src/video/wayland/SDL_waylandwindow.c

```diff
@@ -831,8 +895,15 @@ decoration_frame_configure(struct libdecor_frame *frame,
         height = window->windowed.h;
         wind->floating_resize_pending = SDL_FALSE;
     } else {
+        /*
+         * XXX: When hiding a floating window, libdecor can send bogus content sizes that
+         *      are +/- the height of the title bar, which distorts the window size.
+         *      Ignore any values from libdecor when hiding a floating window.
```

Comment:
Is there a libdecor bug for this?

Reply:
It's linked in the pull request.

https://gitlab.gnome.org/jadahl/libdecor/-/issues/40